CMakeLists.txt

```diff
@@ -10,81 +10,110 @@ set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
-set(MATERIALX_PYTHON_OCIO_DIR "" CACHE PATH
-    "Path to a folder containing the default OCIO configuration to be packaged with MaterialX Python (e.g. 'D:/Projects/OpenColorIO-Configs/aces_1.0.3').")
-set(MATERIALX_PYTHON_PYBIND11_DIR "${CMAKE_CURRENT_SOURCE_DIR}/source/PyMaterialX/PyBind11" CACHE PATH
-    "Path to a folder containing the PyBind11 source. Defaults to the included PyBind11 source, which has been extended to support Python 2.6.")
-
-# Settings to define installation layout    
-set (MATERIALX_INSTALL_INCLUDE_PATH "include" CACHE STRING 
-    "Install header include path (e.g. 'inc', 'include').")
-set (MATERIALX_INSTALL_LIB_PATH "lib" CACHE STRING 
-    "Install lib path (e.g. 'libs', 'lib').")
-set (MATERIALX_INSTALL_STDLIB_PATH "libraries" CACHE STRING 
-    "Install path for mtlx std libs (e.g. 'libraries').")    
-
-mark_as_advanced(MATERIALX_BUILD_DOCS)
-mark_as_advanced(MATERIALX_PYTHON_LTO)
-mark_as_advanced(MATERIALX_INSTALL_PYTHON)
-mark_as_advanced(MATERIALX_WARNINGS_AS_ERRORS)
-mark_as_advanced(MATERIALX_DEBUG_POSTFIX)
-mark_as_advanced(MATERIALX_PYTHON_VERSION)
-mark_as_advanced(MATERIALX_PYTHON_EXECUTABLE)
-mark_as_advanced(MATERIALX_PYTHON_OCIO_DIR)
-mark_as_advanced(MATERIALX_PYTHON_PYBIND11_DIR)
-
 # Shading language generation options
+option(MATERIALX_BUILD_GEN_GLSL "Build the GLSL shader generator back-end." ON)
+option(MATERIALX_BUILD_GEN_OSL "Build the OSL shader generator back-end." ON)
 option(MATERIALX_BUILD_GEN_OGSXML "Build the OGSXML shader generator back-end." ON)
-mark_as_advanced(MATERIALX_BUILD_GEN_OGSXML)
 option(MATERIALX_BUILD_GEN_OGSFX "Build the OGSFX shader generator back-end." ON)
-mark_as_advanced(MATERIALX_BUILD_GEN_OGSFX)
 option(MATERIALX_BUILD_GEN_ARNOLD "Build the Arnold OSL shader generator back-end." ON)
-mark_as_advanced(MATERIALX_BUILD_GEN_ARNOLD)
 
-# Disable render module building option
-option(MATERIALX_DISABLE_BUILD_RENDER "Disable building render modules." OFF)
-mark_as_advanced(MATERIALX_DISABLE_BUILD_RENDER)
+option(MATERIALX_BUILD_PYTHON "Build the MaterialX Python package from C++ bindings. Requires Python 2.7 or greater." OFF)
+option(MATERIALX_BUILD_RUNTIME "Build the MaterialX Runtime data model library." OFF)
+option(MATERIALX_BUILD_RENDER "Build the MaterialX Render modules." ON)
+option(MATERIALX_BUILD_CROSS "Build MaterialXCross folder." OFF)
+
+if(MATERIALX_DISABLE_BUILD_RENDER)
+    message(WARNING "MATERIALX_DISABLE_BUILD_RENDER is deprecated. Please use MATERIALX_BUILD_RENDER instead")
+    set(MATERIALX_BUILD_RENDER OFF)
+endif()
+
+option(MATERIALX_PYTHON_LTO "Enable link-time optimizations for MaterialX Python." ON)
+option(MATERIALX_INSTALL_PYTHON "Install the MaterialX Python package as a third-party library when the install target is built." ON)
 
-# Validation options for compiling and rendering generated shaders
 option(MATERIALX_TEST_RENDER "Run rendering tests for MaterialX Render module. GPU required for graphics validation." OFF)
```

Comment:
@tgvarik , as part of the last ILM merge, this should be on all of the time now, but off in CI.

Reply:
Fixed